### PR TITLE
nodejs: init at 22.0.0, 20.12.2, 18.20.2

### PR DESCRIPTION
**Pull request by Sorixelle**
Kickstarting some of the Javascript work with a pretty obvious (imo) first step - packaging Node.js.

Currently sticking with the 3 versions packaged by Nixpkgs for now. This covers all currently supported versions of Node except v21 which I don't really want to bother with as it's EOL very soon anyway. There's also `latest` and `lts` aliases for their respective versions. Some of the files in here are currently unused - I figure we might want to use them later on though, so I've kept them for now.

Also threw in a simple .gitignore, just to cover a few bits that were cluttering my workdir.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -20,7 +20,14 @@
       packages = forAllSystems (
         { nixPackages, ... }:
         {
-          test = nixPackages.hello;
+          nodejs = rec {
+            v18 = nixPackages.callPackage ./runtimes/nodejs/v18.nix { };
+            v20 = nixPackages.callPackage ./runtimes/nodejs/v20.nix { };
+            v22 = nixPackages.callPackage ./runtimes/nodejs/v22.nix { };
```
Comment (isabelroses) [inline flake.nix]: shall we try using `packagesFromDirectoryRecursive`? https://github.com/auxolotl/core/blob/cbe7a41cc89ccda6bfd14da7c2a8cf982ca07d24/lib/filesystem.nix#L189

Comment (Sorixelle) [reply flake.nix]: Perhaps. I'm not immediately seeing a particularly pretty way to handle `lts` and `latest` aliases with that, though.

Comment (Sorixelle) [reply flake.nix]: Actually - idea. Lemme push something.

Comment (Sorixelle) [reply flake.nix]: Just pushed something a little more refined, how do we feel about this?

**Review by isabelroses — APPROVED** [review]

**Pull request merged**